### DISCUSSION TITLE
Not getting CustomValues in VM Object

### DISCUSSION THREAD

**Discussion by cksidharthan** [vmware/govmomi#3369] [Community Help]
I'm trying to get the "customValue" of a VM, by passing it as props when I'm retreiving Virtual Machines from the vCenter. But I see that the field is coming as nil. I tried getting the vm without passing props so that I'll get all the custom values, but still, That specific field is coming as nil, Is there anything I need to check in the vCenter or is it the way it is supposed to work?

```
var vms []mo.VirtualMachine
err := c.ContainerView.Retrieve(ctx, []string{"VirtualMachine"}, nil, &vms)
```

**Comment by atc0005** [comment]
I don't know whether there are easier ways to accomplish the task, but here is what I ended up using for my needs:

- https://github.com/atc0005/check-vmware/blob/de7a406a53f81cc4e57129d056273f6b2c7e561c/internal/vsphere/attributes.go
